### PR TITLE
Added statistic

### PR DESCRIPTION
**Pull request by Kaspianturbo**

### CODE REVIEW

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Data/Repositories/StatisticRepository.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        public async Task<UsersStatistic> GetUsersStatistic()
+        {
+            var groupStr = "{\"_id\": \"$roles\", \"count\": {$sum: 1}}";
+
+            var aggregate = await _users.Aggregate().Group(groupStr).ToListAsync();
+
+            var statistic = new UsersStatistic();
+            foreach (var doc in aggregate)
+            {
+                var id = doc.GetValue("_id").AsInt32;
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Data/Repositories/StatisticRepository.cs:66]: It is better to convert to RoleOption here

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Data/Repositories/StatisticRepository.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                {
+                    case (int) RoleOption.Employee:
+                        statistic.EmployeesTotal = count;
+                        break;
+                    case (int) RoleOption.Moderator:
+                        statistic.ModeratorsTotal = count;
+                        break;
+                    case (int)RoleOption.Administrator:
+                        statistic.AdministratorsTotal = count;
+                        break;
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Data/Repositories/StatisticRepository.cs:79]: Add default behavior

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.WebApi/Controllers/DiscountsController.cs [hunk 3]
```diff
@@ -105,6 +107,13 @@ public async Task<IActionResult> GetDiscount([FromRoute] Guid id, [FromRoute] La
         public async Task<IActionResult> GetDiscounts([FromBody, CustomizeValidator(RuleSet = "SearchCriteria")] SearchCriteria searchCriteria)
         {
             searchCriteria.SearchUserId = ControllerContext.GetUserId();
+            var role = ControllerContext.GetRole().ToRoleOption();
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.WebApi/Controllers/DiscountsController.cs:110]: Can you add method to return role option as enum initially? for ControllerContext

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.WebApi/Extentions/ControllerExtentions.cs [hunk 4]
```diff
@@ -11,5 +11,9 @@ public static Guid GetUserId(this ControllerContext controllerContext)
         {
             return controllerContext.HttpContext.User.Claims.Where(c => c.Type == "sub").Select(k => k.Value).FirstOrDefault().ToGuid(Guid.Empty);
         }
+        public static string GetRole(this ControllerContext controllerContext)
+        {
+            return controllerContext.HttpContext.User.Claims.Where(c => c.Type == "role").Select(k => k.Value).FirstOrDefault();
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.WebApi/Extentions/ControllerExtentions.cs:16]: What happens if at least User or Claims is null or Empty?

**Review by yegorl — CHANGES_REQUESTED** [review]

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Data/Repositories/StatisticRepository.cs (hunk 1 above)
Comment (Kaspianturbo) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Data/Repositories/StatisticRepository.cs:66]: Done!

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Data/Repositories/StatisticRepository.cs (hunk 2 above)
Comment (Kaspianturbo) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Data/Repositories/StatisticRepository.cs:79]: Done!

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.WebApi/Extentions/ControllerExtentions.cs (hunk 4 above)
Comment (Kaspianturbo) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.WebApi/Extentions/ControllerExtentions.cs:16]: NullReferenceExeption. Fixed is!

**Review by yegorl — APPROVED** [review]

**Pull request merged**